src/Microsoft.AspNetCore.Dispatcher/HandlerFactoryBaseServices.cs

```diff
@@ -0,0 +1,12 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNetCore.Dispatcher
+{
+    public abstract class HandlerFactoryBaseServices
+    {
+        public virtual ILoggerFactory LoggerFactory { get; protected set; }
+    }
```

Comment:
We can add additional services to this without it being a binary breaking change. Anyone that's mocking this for tests will need to respond if their tests cover those code paths.